### PR TITLE
Test and fix headers support

### PR DESCRIPTION
**Pull request by kriskowal**
This adds support for request headers and exercises the tests in the unit test suite. The tests are also available to crossdock, but not this branch does not include the crossdock integration.

r @yarpc/yarpc @yarpc/node

### CODE REVIEW

**Comment by junchaowu** [comment]
I image this one has dealt with cases of headers

File: test/behaviors.js [hunk 1]
```diff
@@ -227,6 +232,121 @@ function run(callback) {
...
+
+        callback(null, new EqualsAssertion(
+            response.headers[token],
+            token,
+            'Server echoed header: ' + token
+        ));
+    }
+}
+
+function headerTestThrift(subject, callback) {
```
Comment (breerly) [inline test/behaviors.js]: these 3 tests look really similar - you think about parameterizing them somehow

File: test/tchannel-raw-inbound.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ test('tchannel raw echo works perfectly', function t(assert) {
     function validate(err, res) {
         assert.ifError(err, 'no error');
         var headers = headersrw.fromBuffer(res.arg2);
-        assert.deepEquals(headers.value, {polo: 'Hello, World!'}, 'headers echoed');
+        assert.deepEquals(headers.value, {marco: 'Hello, World!'}, 'headers echoed');
```
Comment (breerly) [inline test/tchannel-raw-inbound.js]: marco :grinning:

File: transport/http/inbound.js [hunk 3]
```diff
@@ -193,11 +194,21 @@ HttpInbound.prototype.onResponse = function onResponse(err, request, response) {
...
-    request.transportResponse.end(buffer);
+
+    var res = request.transportResponse;
+    res.setHeader('Content-Type', request.handler.encoder.httpResponseContentType);
+
+    if (response.headers) {
+        var headerNames = Object.keys(response.headers);
+        for (var i = 0; i < headerNames.length; i++) {
+            var headerName = headerNames[i];
+            res.setHeader(headerName, response.headers[headerName]);
```
Comment (breerly) [inline transport/http/inbound.js]: nice

**Comment by breerly** [comment]
:shipit:

**Comment by kriskowal** [comment]
@junchaowu There is further work to deal with case conventions.

**Comment by junchaowu** [comment]
lgtm

**Pull request merged**